pkg/monitoring/mimir/utils.go

```diff
@@ -0,0 +1,62 @@
+package mimir
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	"github.com/giantswarm/observability-operator/pkg/monitoring/prometheusagent"
```

Comment:
```suggestion
```

Reply:
So first, you're nor using it as prometheusagent.readRemoteWritePasswordFromSecret so it's not using the one you think it is and I guess your cyclic issue caused the compiler to crash, also this function is private right now (starting with a lower case letter)

Reply:
I changed the function into a public one

Reply:
I still have the `BuildIngressSecret` function as "undefined" though :/

Reply:
Solved it by calling the "ingress" package